beacon-chain/blockchain/head.go

```diff
@@ -142,6 +156,50 @@ func (s *Service) saveHead(ctx context.Context, headRoot [32]byte) error {
 		return errors.Wrap(err, "could not save head root in DB")
 	}
 
+	// Forward an event capturing a new chain head over a common event feed
+	// done in a goroutine to avoid blocking the critical runtime main routine.
+	go func() {
+		dutyEpoch, err := helpers.SlotToEpoch(newHeadSlot).SafeSub(1)
+		if err != nil {
+			log.WithError(err).Error("Could not get duty dependent epoch")
+			return
+		}
+		var previousDutyDependentRoot []byte
+		var currentDutyDependentRoot []byte
+		dutySlot, err := helpers.StartSlot(dutyEpoch)
+		if err != nil {
+			log.WithError(err).Error("Could not get duty slot")
+			return
+		}
+		// In the case where there would be an underflow, we use the genesis state root.
+		if dutySlot == 0 {
+			previousDutyDependentRoot = s.genesisRoot[:]
+			currentDutyDependentRoot = s.genesisRoot[:]
+		} else {
+			previousDutyDependentRoot, err = helpers.BlockRootAtSlot(newHeadState, dutySlot-1)
+			if err != nil {
+				log.WithError(err).Error("Could not get previous duty dependent root")
+				return
+			}
+			currentDutyDependentRoot, err = helpers.BlockRootAtSlot(newHeadState, dutySlot)
+			if err != nil {
+				log.WithError(err).Error("Could not get current duty dependent epoch")
+			}
+		}
+
+		s.cfg.StateNotifier.StateFeed().Send(&feed.Event{
+			Type: statefeed.NewHead,
+			Data: &ethpbv1.EventHead{
+				Slot:                      newHeadSlot,
+				Block:                     headRoot[:],
+				State:                     newStateRoot,
+				EpochTransition:           helpers.IsEpochEnd(newHeadSlot),
+				PreviousDutyDependentRoot: previousDutyDependentRoot,
+				CurrentDutyDependentRoot:  currentDutyDependentRoot,
+			},
+		})
```

Comment:
I commented this out locally and no test failed. Please add a test for this functionality.